src/libfuncs/cast.rs

```diff
@@ -457,41 +404,33 @@ pub fn build_upcast<'ctx, 'this>(
                         )
                         .ok_or(Error::ParseAttributeError)?,
                         location,
-                    ))
-                    .result(0)?
-                    .into();
+                    ))?;
                 result = is_neg_block
-                    .append_operation(arith::addi(result, prime, location))
-                    .result(0)?
-                    .into();
-                is_neg_block.append_operation(cf::br(final_block, &[result], location));
+                    .append_op_result(arith::addi(result, prime, location))?;
+                is_neg_block.append_op_result(cf::br(final_block, &[result], location))?;
```

Comment:
append_operation was correct before because this op has no results

Reply:
This also applies to arith::some_op() right?

Reply:
no, only to the cf ones